File: openapi-generator/scala-gatling/src/gatling/scala/org/openapitools/client/model/DvirBase.scala

package org.openapitools.client.model


case class DvirBase (
    _authorSignature: Option[DvirBaseAuthorSignature],
    /* Signifies if the defects on the vehicle corrected after the DVIR is done. */
    _defectsCorrected: Option[Boolean],
    /* Signifies if the defects on this vehicle can be ignored. */
    _defectsNeedNotBeCorrected: Option[Boolean],
    /* The id of this DVIR record. */
    _id: Option[Long],
    /* Inspection type of the DVIR. */
    _inspectionType: Option[String],
    /* The mechanics notes on the DVIR. */
    _mechanicNotes: Option[String],
    _mechanicOrAgentSignature: Option[DvirBaseMechanicOrAgentSignature],
    _nextDriverSignature: Option[DvirBaseNextDriverSignature],
    /* The odometer reading in miles for the vehicle when the DVIR was done. */
    _odometerMiles: Option[Long],
    /* Timestamp of this DVIR in UNIX milliseconds. */
    _timeMs: Option[Long],
    /* Defects registered for the trailer which was part of the DVIR. */
    _trailerDefects: Option[List[DvirBaseTrailerDefects]],
    /* The id of the trailer which was part of the DVIR. */
    _trailerId: Option[Integer],
    /* The name of the trailer which was part of the DVIR. */
    _trailerName: Option[String],
    _vehicle: Option[DvirBaseVehicle],
    /* The condition of vechile on which DVIR was done. */
    _vehicleCondition: Option[String],
    /* Defects registered for the vehicle which was part of the DVIR. */
    _vehicleDefects: Option[List[DvirBaseTrailerDefects]]
)
object DvirBase {
    def toStringBody(var_authorSignature: Object, var_defectsCorrected: Object, var_defectsNeedNotBeCorrected: Object, var_id: Object, var_inspectionType: Object, var_mechanicNotes: Object, var_mechanicOrAgentSignature: Object, var_nextDriverSignature: Object, var_odometerMiles: Object, var_timeMs: Object, var_trailerDefects: Object, var_trailerId: Object, var_trailerName: Object, var_vehicle: Object, var_vehicleCondition: Object, var_vehicleDefects: Object) =
        s"""
        | {
        | "authorSignature":$var_authorSignature,"defectsCorrected":$var_defectsCorrected,"defectsNeedNotBeCorrected":$var_defectsNeedNotBeCorrected,"id":$var_id,"inspectionType":$var_inspectionType,"mechanicNotes":$var_mechanicNotes,"mechanicOrAgentSignature":$var_mechanicOrAgentSignature,"nextDriverSignature":$var_nextDriverSignature,"odometerMiles":$var_odometerMiles,"timeMs":$var_timeMs,"trailerDefects":$var_trailerDefects,"trailerId":$var_trailerId,"trailerName":$var_trailerName,"vehicle":$var_vehicle,"vehicleCondition":$var_vehicleCondition,"vehicleDefects":$var_vehicleDefects
        | }
        """.stripMargin
}

File: openapi-generator/clojure/src/samsara_api/specs/dvir_base.clj
(ns samsara-api.specs.dvir-base
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [samsara-api.specs.dvir-base-author-signature :refer :all]
            [samsara-api.specs.dvir-base-mechanic-or-agent-signature :refer :all]
            [samsara-api.specs.dvir-base-next-driver-signature :refer :all]
            [samsara-api.specs.dvir-base-trailer-defects :refer :all]
            [samsara-api.specs.dvir-base-vehicle :refer :all]
            [samsara-api.specs.dvir-base-trailer-defects :refer :all]
            )
  (:import (java.io File)))


(def dvir-base-data
  {
   (ds/opt :authorSignature) dvir-base-author-signature-spec
   (ds/opt :defectsCorrected) boolean?
   (ds/opt :defectsNeedNotBeCorrected) boolean?
   (ds/opt :id) int?
   (ds/opt :inspectionType) string?
   (ds/opt :mechanicNotes) string?
   (ds/opt :mechanicOrAgentSignature) dvir-base-mechanic-or-agent-signature-spec
   (ds/opt :nextDriverSignature) dvir-base-next-driver-signature-spec
   (ds/opt :odometerMiles) int?
   (ds/opt :timeMs) int?
   (ds/opt :trailerDefects) (s/coll-of dvir-base-trailer-defects-spec)
   (ds/opt :trailerId) int?
   (ds/opt :trailerName) string?
   (ds/opt :vehicle) dvir-base-vehicle-spec
   (ds/opt :vehicleCondition) string?
   (ds/opt :vehicleDefects) (s/coll-of dvir-base-trailer-defects-spec)
   })

(def dvir-base-spec
  (ds/spec
    {:name ::dvir-base
     :spec dvir-base-data}))
